src/components/Tabs/AnnotationTab.tsx

```diff
@@ -0,0 +1,360 @@
+    setShowEditPopover(false);
+  };
+
+  const onClickEditSave = (): void => {
+    if (currentLabelIdx !== null) {
+      setLabelName(currentLabelIdx, editPopoverNameInput);
+    }
+    setShowEditPopover(false);
+  };
+
+  const tableColumns: TableProps<TableDataType>["columns"] = [
+    {
+      title: "Object ID",
+      dataIndex: "id",
+      key: "id",
+      sorter: (a, b) => a.id - b.id,
+    },
+    {
+      title: "Track ID",
+      dataIndex: "track",
+      key: "track",
+      sorter: (a, b) => a.track - b.track,
+    },
+    {
+      title: "Time",
+      dataIndex: "time",
+      key: "time",
+      sorter: (a, b) => a.time - b.time,
+    },
+    // Column that contains a remove button for the ID.
+    {
+      title: "",
+      key: "action",
+      render: (_, record) => (
+        <IconButton
+          type="text"
+          onClick={(event) => {
+            // Rows have their own behavior on click (jumping to a timestamp),
+            // so we need to stop event propagation so that the row click event
+            // doesn't fire.
+            event.stopPropagation();
+            setLabelOnId(currentLabelIdx!, record.id, false);
+          }}
+        >
+          <CloseOutlined />
+          <VisuallyHidden>
+            Remove ID {record.id} (track {record.track})
+          </VisuallyHidden>
+        </IconButton>
+      ),
+    },
+  ];
+
+  const tableData: TableDataType[] = useMemo(() => {
+    const dataset = props.dataset;
+    if (currentLabelIdx !== null && dataset) {
+      const ids = annotationData.getLabeledIds(currentLabelIdx);
+      return ids.map((id) => {
+        const track = dataset.getTrackId(id);
+        const time = dataset.getTime(id);
+        return { key: id.toString(), id, track, time };
+      });
+    }
+    return [];
+  }, [annotationData, currentLabelIdx, props.dataset]);
+
+  // Options for the selection dropdown
+  const selectLabelOptions: ItemType[] = labels.map((label, index) => {
+    return {
+      key: index.toString(),
+      label: label.ids.size ? `${label.name} (${label.ids.size})` : label.name,
+    };
+  });
+
+  // A small popup that appears when you press the edit button.
+  const editPopoverContents = (
+    <FlexColumn style={{ width: "250px" }} $gap={10}>
+      <label style={{ gap: "10px", display: "flex", flexDirection: "row" }}>
+        <span>Name</span>
+        <Input
+          value={editPopoverNameInput}
+          onChange={(e) => setEditPopoverNameInput(e.target.value)}
+          ref={editPopoverInputRef}
+        ></Input>
+      </label>
+      <label style={{ gap: "10px", display: "flex", flexDirection: "row" }}>
+        <span>Color</span>
+        <div>
+          <ColorPicker
+            size="small"
+            value={new AntdColor(selectedLabel?.color.getHexString() || "ff00ff")}
+            onChange={(_, hex) => {
+              currentLabelIdx !== null && setLabelColor(currentLabelIdx, new Color(hex as HexColorString));
```

Comment:
take it or leave it: push `currentLabelIdx !== null` check into `setLabelColor` so you can just call it without always having to wrap it in an if statement